pkg/apis/clusterlink.net/v1alpha1/operator_types.go

```diff
@@ -0,0 +1,131 @@
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StatusType represents the status options for ClusterLink components.
+type StatusType string
+
+const (
+	// StatusNone means a resource was not created.
+	StatusNone StatusType = "none"
+	// StatusPending means a resource was created but is not ready to use.
+	StatusPending StatusType = "pending"
+	// StatusError means an error occurred when the resource was created.
+	StatusError StatusType = "error"
+	// StatusReady means a resource was created and is ready to use.
+	StatusReady StatusType = "ready"
+)
+
+// IngressType represents the ingress type  of the deployed ClusterLink.
+type IngressType string
+
+const (
+	// IngressTypeNone indicates that the deployment operator did not create any external ingress.
+	IngressTypeNone IngressType = "none"
+	// IngressTypeNodePort indicates that the deployment operator created an external ingress of type NodePort.
+	IngressTypeNodePort IngressType = "NodePort"
+	// IngressTypeLoadBalancer indicates that the deployment operator created an external ingress of type LoadBalancer.
+	IngressTypeLoadBalancer IngressType = "LoadBalancer"
+)
+
+// DataplaneType represents the dataplane type  of the deployed ClusterLink.
+type DataplaneType string
+
+const (
+	// GoType indicates that the dataplane type is Go-dataplane.
+	GoType DataplaneType = "go"
+	// EnvoyType indicates that the dataplane type is Envoy.
+	EnvoyType DataplaneType = "envoy"
+)
+
+// OperatorSpec defines the desired state of a ClusterLink Operator.
+type OperatorSpec struct {
+	DataPlane DataPlaneSpec `json:"dataplane,omitempty"`
+	Ingress   IngressSpec   `json:"ingress,omitempty"`
+
+	// +kubebuilder:validation:Enum=trace;debug;info;warning;error;fatal
+	// +kubebuilder:default=info
+	// LogLevel define the ClusterLink components log level.
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// +kubebuilder:default="ghcr.io/clusterlink-net"
+	// ContainerRegistry is the container registry to pull the ClusterLink project images.
+	ContainerRegistry string `json:"containerRegistry,omitempty"`
+	// +kubebuilder:default="latest"
+	// ImageTag represents the version of the ClusterLink project images.
+	ImageTag string `json:"imageTag,omitempty"`
+}
+
+// OperatorStatus defines the observed state of ClusterlLink Operator.
+type OperatorStatus struct {
+	Controlplane ComponentStatus `json:"controlplane,omitempty"`
+	Dataplane    ComponentStatus `json:"dataplane,omitempty"`
+}
+
+// ComponentStatus defines the status of component in ClusterLink.
+type ComponentStatus struct {
+	// +kubebuilder:validation:Enum=none;ready;error;pending
+	// +kubebuilder:default=none
+	Status  StatusType `json:"status,omitempty"`
+	Message string     `json:"message,omitempty"`
+}
+
+// DataPlaneSpec defines the desired state of the dataplane components in ClusterLink.
+type DataPlaneSpec struct {
```

Comment:
Especially since you have status for both, Is there a corresponding `ControlPlaneSpec`?

Reply:
For now I don't need it, but probably with the new controlplane I will add it.